Source/Laser/LaserParticleContainer.cpp

```diff
@@ -34,70 +34,27 @@ LaserParticleContainer::LaserParticleContainer (AmrCore* amr_core, int ispecies,
     std::string laser_type_s;
     pp.get("profile", laser_type_s);
     std::transform(laser_type_s.begin(), laser_type_s.end(), laser_type_s.begin(), ::tolower);
-    if (laser_type_s == "gaussian") {
-        profile = laser_t::Gaussian;
-    } else if(laser_type_s == "harris") {
-        profile = laser_t::Harris;
-    } else if(laser_type_s == "parse_field_function") {
-        profile = laser_t::parse_field_function;
-    } else {
-        amrex::Abort("Unknown laser type");
-    }
 
     // Parse the properties of the antenna
     pp.getarr("position", position);
     pp.getarr("direction", nvec);
     pp.getarr("polarization", p_X);
+    stc_direction = p_X;
+    pp.queryarr("stc_direction", stc_direction);
+
+
     pp.query("pusher_algo", pusher_algo);
     pp.get("wavelength", wavelength);
     pp.get("e_max", e_max);
     pp.query("do_continuous_injection", do_continuous_injection);
     pp.query("min_particles_per_mode", min_particles_per_mode);
 
-    if ( profile == laser_t::Gaussian ) {
-        // Parse the properties of the Gaussian profile
-        pp.get("profile_waist", profile_waist);
-        pp.get("profile_duration", profile_duration);
-        pp.get("profile_t_peak", profile_t_peak);
-        pp.get("profile_focal_distance", profile_focal_distance);
-        stc_direction = p_X;
-        pp.queryarr("stc_direction", stc_direction);
-        pp.query("zeta", zeta);
-        pp.query("beta", beta);
-        pp.query("phi2", phi2);
-    }
-
-    if ( profile == laser_t::Harris ) {
-        // Parse the properties of the Harris profile
-        pp.get("profile_waist", profile_waist);
-        pp.get("profile_duration", profile_duration);
-        pp.get("profile_focal_distance", profile_focal_distance);
-    }
-
-    if ( profile == laser_t::parse_field_function ) {
-        // Parse the properties of the parse_field_function profile
-        pp.get("field_function(X,Y,t)", field_function);
-        parser.define(field_function);
-        parser.registerVariables({"X","Y","t"});
-
-        ParmParse ppc("my_constants");
-        std::set<std::string> symbols = parser.symbols();
-        symbols.erase("X");
-        symbols.erase("Y");
-        symbols.erase("t"); // after removing variables, we are left with constants
-        for (auto it = symbols.begin(); it != symbols.end(); ) {
-            Real v;
-            if (ppc.query(it->c_str(), v)) {
-                parser.setConstant(*it, v);
-                it = symbols.erase(it);
-            } else {
-                ++it;
-            }
-        }
-        for (auto const& s : symbols) { // make sure there no unknown symbols
-            amrex::Abort("Laser Profile: Unknown symbol "+s);
-        }
+    //Select laser profile
+    if(laser_profiles_dictionary.count(laser_type_s) == 0){
+        amrex::Abort("Unknown laser type");
```

Comment:
Potentially more expressive for users:
```suggestion
        amrex::Abort(std::string("Unknown laser type: ").append(laser_type_s));
```